### PR TITLE
[BEAM-5758] add fanout and reiteration to LoadTests

### PR DESCRIPTION
**Pull request by kkucharc**
In GBK and Combine load tests there were missing both options: fanout and reiteration. Both are parametrised and can be passed by pipeline options: `fanout` and `iterations`

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | --- | --- | --- | ---
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by kkucharc** [comment]
Run Python Load Tests Smoke

**Comment by kkucharc** [comment]
R: @lgajowy could you take a look at it? cc: @pabloem

File: sdks/python/apache_beam/testing/load_tests/group_by_key_test.py [hunk 1]
```diff
@@ -166,18 +188,31 @@ def setUp(self):
...
+                 self.parseTestPipelineOptions()))
+             | 'Measure time: Start' >> beam.ParDo(
+                 MeasureTime(self.metrics_namespace))
+            )
+
+    for branch in range(self.fanout):
+      # pylint: disable=expression-not-assigned
+      (input
+       | 'GroupByKey %i' % branch >> beam.GroupByKey()
+       | 'Ungroup %i' % branch >> beam.FlatMap(
```
Comment (lgajowy) [inline sdks/python/apache_beam/testing/load_tests/group_by_key_test.py]: It should use `UngroupAndReiterate` here, right?

Comment (kkucharc) [reply sdks/python/apache_beam/testing/load_tests/group_by_key_test.py]: You are right! I fixed this

**Comment by kkucharc** [comment]
Run Python Load Tests Smoke

**Comment by kkucharc** [comment]
I waited for tests to pass, @lgajowy can you take a look if now everything is correct?

**Comment by lgajowy** [comment]
LGTM

**Pull request merged**